crates/node/src/keyshare.rs

```diff
@@ -289,16 +288,30 @@ impl KeyshareStorage {
 
     /// Imports keyshares from the provided backup into permanent storages.
     ///
+    /// The following validation is performed:
+    /// - the provided backup shares [`Keyshare`] must be an exact match for the provided conrtact_keyset [`Keyset`]
+    /// - if the **permanent** KeyshareStorage contains keyshares of the same epoch, then they must be a prefix of the provided backup shares.
+    /// - if the **temporary** KeyshareStorage contains any keyshares of matching [`KeyEventId`],
+    ///   then they must be an exact match for the keyshares provided in the backup.
+    ///
+    /// Any keyshares present missing in the temporary or local Keyshare storage are drawn from the
+    /// provided backup.
+    ///
+    /// If the validation passes, then the constructed keyset is stored to permanent
+    /// KeyshareStorage and any temporary keyshares of younger epoch ids are permanently deleted.
+    ///
     /// # Errors
     /// Returns an error if:
     /// - The backup does not match the contract keyset’s epoch or domains,
-    /// - The permanent keyshare storage is not empty.
+    /// - A required keyshare is missing in both temporary storage and backup,
+    /// - The reconstructed keyshares differ from the backup (indicating corruption),
+    /// - Or storing to permanent storage fails.
     ///
     /// # Returns
     /// * `Ok(())` if the backup was successfully imported and stored permanently.
     /// * `Err(anyhow::Error)` if any validation or storage step fails.
-    #[allow(dead_code)] // todo: remove after integration with onboarding function
     pub async fn import_backup(
+        // while technically not required to be mut, we must not call read functions in parallel
```

Comment:
Isn't this a write function?
```suggestion
        // while technically not required to be mut, we must not call write functions in parallel
```

Otherwise it's nice to use the borrow checker here to guard against concurrent writes.